workspace/apps/pidp/src/app/features/organization-info/pages/endorsements/endorsements.page.ts

```diff
@@ -211,6 +213,37 @@ export class EndorsementsPage
       .subscribe();
   }
 
+  public getStatus(endorsementRequestStatus: EndorsementRequestStatus): string {
+    return (
+      endorsementRequestsLabelText[endorsementRequestStatus] ?? 'Requested'
+    );
+  }
+
+  public isEndorsementRequested(
+    endorsementRequestStatus: EndorsementRequestStatus,
+  ): boolean {
+    return ![
+      EndorsementRequestStatus.CANCELLED,
+      EndorsementRequestStatus.DECLINED,
+    ].includes(endorsementRequestStatus);
+  }
+
+  public getCollegeTextForEndorsement(endorsement: Endorsement): string {
+    const college = this.lookupService.colleges.find(
+      (x) => x.code === endorsement.collegeCode,
+    );
+    return college?.name ?? '';
+  }
+
+  public getCollegeTextForEndorsementRequest(
```

Comment:
the functions `getCollegeTextForEndorsement`  and `getCollegeTextForEndorsementRequest` doing the same functionality but only input type is different.  I think we can make it one function with the union types.
` public getCollegeTextForEndorsement(endorsement: Endorsement | EndorsementRequest): string `.
With this function signature one function will serve the required funtionality.
reference link : [EndorsementRequest](url)

Reply:
nice catch Vinod!